cli/js/40_jupyter.js

```diff
@@ -402,19 +402,42 @@ function enableJupyter() {
   const { op_jupyter_broadcast, op_jupyter_comm_recv, op_jupyter_comm_open } =
     core.ops;
 
-  function commOpen(commId, targetName) {
+  function comm(commId, targetName, msgCallback) {
     op_jupyter_comm_open(commId, targetName);
-  }
 
-  async function commRecv(commId) {
-    const [data, buffers] = await op_jupyter_comm_recv(commId);
+    let closed = false;
+
+    // TODO(bartlomieju): return something, so we can close this comm.
+    (async () => {
+      while (true) {
+        const [data, buffers] = await op_jupyter_comm_recv(commId);
+
+        if (!data) {
+          closed = true;
+          break;
+        }
+
+        msgCallback?.({
+          ...data,
+          buffers,
+        });
+      }
+    })();
 
-    if (!data) {
-      return undefined;
-    }
     return {
-      ...data,
-      buffers,
+      close() {
+        if (closed) {
```

Comment:
We also need to use an `op_jupyter_` I assume, to clean up the mapping on the Rust side.